diff --git a/android/src/play/java/com/dooboolab/rniap/RNIapModule.kt b/android/src/play/java/com/dooboolab/rniap/RNIapModule.kt
index 54ca6969c543f5826d77b4ff8cdfcf960cbb5ffa..58d6bf53d43151a36b95dd361c72682ae1196c4c 100644
--- a/android/src/play/java/com/dooboolab/rniap/RNIapModule.kt
+++ b/android/src/play/java/com/dooboolab/rniap/RNIapModule.kt
@@ -461,7 +461,7 @@ class RNIapModule(
         isOfferPersonalized: Boolean, // New parameter in V5
         promise: Promise,
     ) {
-        val activity = currentActivity
+        val activity = reactApplicationContext.currentActivity
         if (activity == null) {
             promise.safeReject(PromiseUtils.E_UNKNOWN, "getCurrentActivity returned null")
             return
diff --git a/RNIap.podspec b/RNIap.podspec
index 0b8a4ea6b6da6a9f025b66142b23c528fe06f6d6..ac98a77b1d3809d2e13bbf15fb1ff317e94c449f 100644
--- a/RNIap.podspec
+++ b/RNIap.podspec
@@ -23,7 +23,6 @@ Pod::Spec.new do |s|
   # Don't install the dependencies when we run `pod install` in the old architecture.
   if ENV['RCT_NEW_ARCH_ENABLED'] == '1' then
-    s.compiler_flags = folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED=1"
-    s.pod_target_xcconfig    = {
+    pod_target_xcconfig = {
         "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/boost\"",
         "CLANG_CXX_LANGUAGE_STANDARD" => "c++17"
     }
@@ -31,4 +30,12 @@ Pod::Spec.new do |s|
-    s.dependency "RCT-Folly"
+    if ENV['RCT_USE_RN_DEP'] == '1'
+      pod_target_xcconfig["HEADER_SEARCH_PATHS"] = "#{pod_target_xcconfig["HEADER_SEARCH_PATHS"]} \"$(PODS_ROOT)/ReactNativeDependencies\""
+      s.dependency "ReactNativeDependencies"
+    else
+      s.dependency "RCT-Folly"
+    end
+
+    s.compiler_flags = folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED=1"
+    s.pod_target_xcconfig = pod_target_xcconfig
     s.dependency "RCTRequired"
     s.dependency "RCTTypeSafety"
     s.dependency "ReactCommon/turbomodule/core"
